frontend/test/pytest/test_cuda_integration.py

```diff
@@ -476,6 +480,97 @@ def circuit2(a):
+        dev2 = qml.device("lightning.qubit", wires=2)
+
+        def circuit(params):
+            x, y = jax.numpy.array_split(params, 2)
+            qml.RX(x[0], wires=[0])
+            qml.RX(y[0], wires=[0])
+            return qml.expval(qml.PauliZ(0))
+
+        circuit1 = catalyst.third_party.cuda.cudaqjit(qml.QNode(circuit, dev1))
+        circuit2 = qjit(qml.QNode(circuit, dev2))
+        spy = mocker.spy(circuit1, "capture")
+
+        p = jnp.array([0.1, 0.2])
+        res1 = circuit1(p)
+        spy.assert_called()
+        assert_allclose(res1, circuit2(p))
+
+        p = jnp.array([0.3, 0.4])
+        spy = mocker.spy(circuit1, "capture")
+        res2 = circuit1(p)
+        spy.assert_not_called()
+        assert_allclose(res2, circuit2(p))
+
+    def test_aot_capture(self, mocker):
+        """Test that JAXPR capture can occur AOT"""
+        dev1 = qml.device("softwareq.qpp", wires=2)
+        dev2 = qml.device("lightning.qubit", wires=2)
+
+        def circuit(x: float, y: float):
+            qml.RX(x, wires=[0])
+            qml.RX(y, wires=[0])
+            return qml.expval(qml.PauliZ(0))
+
+        circuit1 = catalyst.third_party.cuda.cudaqjit(qml.QNode(circuit, dev1))
+        circuit2 = qjit(qml.QNode(circuit, dev2))
+
+        x, y = 0.1, 0.2
+        spy = mocker.spy(circuit1, "capture")
+        res1 = circuit1(x, y)
+        spy.assert_not_called()
+        assert_allclose(res1, circuit2(x, y))
+
+        x, y = 0.3, 0.4
+        res2 = circuit1(x, y)
+        spy.assert_not_called()
+        assert_allclose(res2, circuit2(x, y))
+
+    def test_autograph(self):
+        """Test that autograph can be invoked"""
+        dev = qml.device("softwareq.qpp", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x: float, y: float):
+
+            for _ in range(10):
+                qml.RX(x, wires=[0])
+
+            qml.RX(y, wires=[0])
+            return qml.state()
+
+        circuit1 = catalyst.third_party.cuda.cudaqjit(circuit, autograph=True)
+        assert "for_loop" in str(circuit1.jaxpr)
+
+        circuit2 = catalyst.third_party.cuda.cudaqjit(circuit, autograph=False)
+        assert "for_loop" not in str(circuit2.jaxpr)
+
+    @pytest.mark.skip(reason="kwargs currently not supported")
+    def test_kwargs(self):
+        """Test passing kwargs to an qjit"""
+        dev1 = qml.device("softwareq.qpp", wires=2)
+        dev2 = qml.device("lightning.qubit", wires=2)
+
+        def circuit(x, y=0.2):
+            qml.RX(x, wires=[0])
+            qml.RX(y, wires=[0])
+            return qml.expval(qml.PauliZ(0))
+
+        circuit1 = catalyst.third_party.cuda.cudaqjit(qml.QNode(circuit, dev1))
+        circuit2 = qjit(qml.QNode(circuit, dev2))
+
+        # test using default values
+        res1 = circuit1(0.1)
+        assert_allclose(res1, circuit2(0.1))
+
+        # test passing kwargs
+        res1 = circuit1(x=0.1, y=0.3)
+        assert_allclose(res1, circuit2(x=0.1, y=0.3))
+
```

Comment:
What happens if you use keep intermediate?

Reply:
Nothing. Just tried it. :+1: